assembl/static2/js/app/components/common/modal.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { Link } from 'react-router';
+import { Modal } from 'react-bootstrap';
+
+class AssemblModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: false
+    };
+    this.close = this.close.bind(this);
+    this.open = this.open.bind(this);
+  }
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      showModal: nextProps.showModal
+    });
+  }
+  close() {
+    this.setState({
+      showModal: false
+    });
+  }
+  open() {
+    this.setState({
+      showModal: true
+    });
+  }
+  render() {
+    const { title, body, footer, link } = this.props;
+    return (
+      <Modal show={this.state.showModal} onHide={this.close}>
+        <Modal.Header closeButton>
+          {title &&
+            <Modal.Title>{ title }</Modal.Title>
+          }
+        </Modal.Header>
+        {body &&
+          <Modal.Body>
+            <div>{ body }</div>
+          </Modal.Body>
+        }
+        {footer &&
+          <Modal.Footer>
+            <Link to={link} className="button-link button-dark">{ footer }</Link>
```

Comment:
Another way, after the merge with my PR, is to create a closure method in Routes object, and pass that along to the component. Then the component can check if it is a function, then it's an internal route, if link is just a string, it's an external route.
eg.
```
Routes.prototype.getRoute = (path) => { return (args) => Routes.get(route); };
<Modal link={Routes.getRoute('someInternalPath')} />
```
In modal:
```
if (_.isFunc(this.props.link)){ return (<Link to={ this.props.link() } /> ) } else { return ( <a href={this.props.link()}></a> )}
```